openthread/src/srp.rs

```diff
@@ -200,6 +198,23 @@ impl Display for SrpState {
     }
 }
 
+#[cfg(feature = "defmt")]
+impl defmt::Format for SrpState {
```

Comment:
Maybe we can do something like https://github.com/esp-rs/esp-hal/blob/1db73628c24f6c9c7d3938b233ceecabee42fe31/esp-hal/src/twai/mod.rs#L174-L203 to avoid duplicating all the strings. What do you think?

Reply:
Hm, or maybe not? Look at the `Other(_)` member.